stream-chat-android/src/main/java/com/getstream/sdk/chat/view/ChannelHeaderViewStyle.kt

```diff
@@ -19,75 +21,138 @@ internal class ChannelHeaderViewStyle(context: Context?, attrs: AttributeSet?) :
     val optionsButtonWidth: Int
     val optionsButtonHeight: Int
     val isActiveBadgeShow: Boolean
-    val backButtonBackground: Drawable
-    val optionsButtonBackground: Drawable
+    val backButtonBackground: Drawable?
+    val optionsButtonBackground: Drawable?
     val offlineText: String
     val channelWithoutNameText: String
+    val avatarStyle: AvatarStyle = AvatarStyle()
+
+    private val res = context.resources
 
     init {
-        // parse the attributes
-        setContext(context)
-        val attributes = getContext().obtainStyledAttributes(
+        val attributes = context.obtainStyledAttributes(
             attrs,
             R.styleable.ChannelHeaderView,
             0,
             0
         )
 
         channelTitleText = TextStyle.Builder(attributes).apply {
-            size(R.styleable.ChannelHeaderView_streamChannelHeaderTitleTextSize, getDimension(R.dimen.stream_channel_header_initials))
-            color(R.styleable.ChannelHeaderView_streamChannelHeaderTitleTextColor, getColor(R.color.stream_channel_initials))
+            size(
+                R.styleable.ChannelHeaderView_streamChannelHeaderTitleTextSize,
+                res.getDimensionPixelSize(R.dimen.stream_channel_header_initials)
+            )
```

Comment:
I know you didn't introduce the pattern, but we really shouldn't need to use `apply` here, the point of the `Builder` we have is to provide a nice API for chaining calls.